### PR TITLE
List documents also works for kvk and paginates for documentlist

### PR DESCRIPTION
**Pull request by drakonen**

### CODE REVIEW

File: decosjoin/api/decosjoin/decosjoin_connection.py [hunk 1]
```diff
@@ -178,20 +178,45 @@ def filter_zaken(self, zaken):
...
+
+    def get_all_pages(self, url):
+        """ Get 'content' from all pages for the provided url """
+
+        if '?' in url:  # 'proper' url arguments...
+            url = f'{url}&top={page_size}'
+        else:
+            url = f'{url}?top={page_size}'
+        items = []
+        # fetch one. to get the count
```
Comment (timvanoostrom) [inline decosjoin/api/decosjoin/decosjoin_connection.py:200]: Wat bedoel je met _fetch one_ ?

File: decosjoin/api/decosjoin/decosjoin_connection.py [hunk 2]
```diff
@@ -178,20 +178,45 @@ def filter_zaken(self, zaken):
...
+        if log_raw:
+            from pprint import pprint
+            print("request:", url)
+            pprint(res_json)
+        return res_json
+
+    def get_all_pages(self, url):
+        """ Get 'content' from all pages for the provided url """
+
+        if '?' in url:  # 'proper' url arguments...
```
Comment (timvanoostrom) [inline decosjoin/api/decosjoin/decosjoin_connection.py:195]: Voor de mooi kun je de url misschien beter parsen en weer samenstellen

File: decosjoin/tests/test_api.py [hunk 3]
```diff
@@ -57,6 +64,7 @@ def test_status(self):
 
     @patch("decosjoin.server.DecosJoinConnection._get_response", get_response_mock)
     @patch("decosjoin.server.DecosJoinConnection._post_response", post_response_mock)
+    @patch('decosjoin.api.decosjoin.decosjoin_connection.page_size', 10)
     def test_getvergunningen(self):
         response = self._client_get("/decosjoin/getvergunningen")
         self.assertEqual(response.status_code, 200, response.data)
```
Comment (timvanoostrom) [inline decosjoin/tests/test_api.py:70]: Passing response.data is for debugging I presume, correct?

**Review by timvanoostrom — COMMENTED** [review]
How does the pagination work from the front-end? Is page_size 10 enought to get all documents?

File: decosjoin/tests/test_api.py (hunk 3 above)
Comment (drakonen) [reply decosjoin/tests/test_api.py:70]: correct

**Review by timvanoostrom — APPROVED** [review]

**Pull request merged**